### ISSUE TITLE
Learning rate scheduler

### ISSUE THREAD

**Issue by lhq12** [TsingZ0/PFL-Non-IID#99]
你好，非常感谢你能开源这个有意义的项目。不过我发现，在调整lr的时候，并非全局统一调整，那么在只选择一部分client参与训练的情况下，就会出现同一时刻，不同client的lr不一致的问题，这会不会对训练过程造成什么影响？

**Comment by TsingZ0** [comment]
感谢你的提议，目前来看确实有这个问题，所以目前`lr_scheduler`但是个可选项。如果同步的话，那就要求client每一轮都进行通讯了，那是不是违背了client部分参与的setting呢？

欢迎就此问题进行讨论，并期待你的宝贵意见~

**Comment by lhq12** [comment]
我觉得可以把 lr_scheduler放在server，在每轮通信开始时，由server同步lr；或者server端告知client当前的round_idx，然后直接调用函数，根据初始lr和当前round_idx计算lr

**Comment by TsingZ0** [comment]
感谢你的建议，感觉很不错

> 在每轮通信开始时，由server同步lr

这样的话，那就要求client每一轮都参与通讯了

> server端告知client当前的round_idx，然后直接调用函数，根据初始lr和当前round_idx计算lr

这样的话，不规律的lr变化，会影响模型收敛性吗？如果某个client长期不参与，它的下次lr会变得非常小。这么说的话，其实还涉及具体client sampling的设计。

**Comment by lhq12** [comment]
> 每轮通信开始时，有server同步lr
我理解的是server将新一轮的lr同步给active clients就可以；

> 不规律的lr变化，会影响模型收敛性吗？如果某个client长期不参与，它的下次lr会变得非常小
这个确实是一个值得研究的问题；但是如果每轮更新时，不同client的lr不同，模型的收敛性会是怎么样？而且在这种条件下，聚合时根据样本数量加权平均梯度好像也就没有什么意义了；

**Comment by TsingZ0** [comment]
是的，不过目前大多数论文都默认lr不变吧，也就是没用到lr_scheduler
